src/nginx/config.cc

```diff
@@ -593,7 +602,15 @@ ngx_int_t ngx_esp_build_server_config(ngx_conf_t *cf, ngx_esp_loc_conf_t *lc,
 
   // validate rewrite rule syntax
   if (config.has_api_service_config()) {
-    std::unique_ptr<std::regex> rule_check;
+    pcre *regex_compiled_;
+    pcre_extra *regex_extra_;
+
+    // backup functional pointers of pcre memory functions
+    origin_pcre_malloc = pcre_malloc;
```

Comment:
please try to re-use the code in rewrite.cc

Reply:
may be move these verification code into rewrite.cc?

Reply:
Created a static method, RewriteRule::ValidateRewriteRule